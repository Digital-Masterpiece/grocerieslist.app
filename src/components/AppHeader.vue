<template>
  <header class="header">
    <nav aria-label="Primary" class="header__nav">
      <router-link to="'/" title="View my Lists" class="header__link">
        <font-awesome-icon icon="list-alt" class="header__link__icon"/>
        <span>My Lists</span>
      </router-link>
      <router-link to="/new" title="Create a new List" class="header__link header__link--new-list">
        <font-awesome-icon icon="plus-circle" class="header__link__icon"/>
        <span>New List</span>
      </router-link>
    </nav>
  </header>
</template>

<style lang="scss">
.header {
  @apply flex justify-center items-center flex-shrink-0 p-6 text-xl w-full bg-gray-800 border-b border-gray-600;

  &__nav {
    @apply flex justify-between items-center w-full max-w-lg;
  }

  &__link {
    @apply flex justify-center items-center;
    &__icon {
      @apply mr-2 text-2xl;
    }
    &--new-list {
      @apply ml-auto;
    }
  }
}
</style>