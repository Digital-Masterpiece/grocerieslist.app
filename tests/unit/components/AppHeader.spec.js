import {shallowMount,} from '@vue/test-utils'
import AppHeader from '@/components/AppHeader'
import {FontAwesomeIcon} from '@fortawesome/vue-fontawesome'

const factory = () => {
    return shallowMount(AppHeader, {
        stubs: [
            FontAwesomeIcon,
            'router-link'
        ]
    });
}

describe('New Item Form', () => {
    it('Renders Properly', () => {
        const wrapper = factory()
        expect(wrapper).toMatchSnapshot()
    })
})