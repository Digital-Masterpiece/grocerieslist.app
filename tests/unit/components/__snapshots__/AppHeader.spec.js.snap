// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Item Form Renders Properly 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__emits": Object {},
      "__props": Array [
        Object {},
        Array [],
      ],
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <header
        class="header"
      >
        <nav
          aria-label="Primary"
          class="header__nav"
        >
          <router-link
            class="header__link"
            title="View my Lists"
            to="'/"
          >
            <font-awesome-icon
              class="header__link__icon"
              icon="list-alt"
            />
            <span>
              My Lists
            </span>
          </router-link>
          <router-link
            class="header__link header__link--new-list"
            title="Create a new List"
            to="/new"
          >
            <font-awesome-icon
              class="header__link__icon"
              icon="plus-circle"
            />
            <span>
              New List
            </span>
          </router-link>
        </nav>
      </header>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
          "props": undefined,
          "render": [Function],
        },
        "transition-group": Object {
          "name": "transition-group",
          "props": undefined,
          "render": [Function],
        },
      },
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__emits": null,
          "__props": Array [],
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {},
      "reload": [Function],
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.5",
  },
  "__setProps": [Function],
  "componentVM": Object {},
  "rootVM": Object {},
}
`;
